okhttp/src/commonJvmAndroid/kotlin/okhttp3/TrailersSource.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (C) 2025 Square, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package okhttp3
+
+import okio.IOException
+
+@ExperimentalOkHttpApi
+fun interface TrailersSource {
+  /** Returns the trailers, blocking if they need to be loaded. */
```

Comment:
There’s still a bug in the way this is wired up.

`Response.trailers()` discards the response body automatically, and then calls this function. What SHOULD be happening is this function itself should discard the response body. It’s not wired up that way because it’s a bigger change, but I might attempt that in a follow-up.

Not urgent for 5.0!

Reply:
What's the actual bug apart from code in the wrong place? Is it observable?